logprep/processor/generic_adder/processor.py

```diff
@@ -43,15 +56,131 @@ def __init__(self, name: str, skipped_fields: List[str]):
         super().__init__(name, message)
 
 
+def sql_config_validator(_, attribute, value):
+    """validate if a subfield of a dict is valid"""
+    if attribute.name == "sql_config" and isinstance(value, dict):
+        table = value.get("table")
+        if table and re.search(r"\s", table):
+            raise InvalidConfigurationError(f"Table in 'sql_config' contains whitespaces!")
+        if table and not re.search(r"^[a-zA-Z0-9_]+$", table):
+            raise InvalidConfigurationError(
+                f"Table in 'sql_config' may only contain alphanumeric characters and underscores!"
+            )
+
+
 class GenericAdder(Processor):
     """Resolve values in documents by referencing a mapping list."""
 
-    rule_class = GenericAdderRule
+    @define(kw_only=True)
+    class Config(Processor.Config):
+        """GenericAdder config"""
 
-    def _apply_rules(self, event, rule):
-        conflicting_fields = list()
+        sql_config: Optional[dict] = field(
+            default=None,
+            validator=[
+                validators.optional(validator=validators.instance_of(dict)),
+                sql_config_validator,
+            ],
+        )
+        """
+        Configuration of the connection to a MySQL database and settings on how to add data from
+        the database.
+        This field is optional. The database feature will not be used if `sql_config` is omitted.
+        Has following subfields:
+
+        - `user` - The user to use when connecting to the MySQL database.
+        - `password` - The password to use when connecting to the MySQL database.
+        - `host` - The host to use when connecting to the MySQL database.
+        - `database` - The database name to use when connecting to the MySQL database.
+        - `table` - The table name to use when connecting to the MySQL database.
+        - `target_column` - The name of the column whose values are being matched against a value
+          from an event.
+          If a value matches, the remaining values of the row with the match are being added to
+          the event.
+        - `add_target_column` - Determines if the target column itself will be added to the event.
+          This is set to false per default.
+        - `timer` - Period how long to wait (in seconds) before the database table is being checked
+          for changes.
+          If there is a change, the table is reloaded by Logprep.
+        """
 
-        for dotted_field, value in rule.add.items():
+    rule_class = GenericAdderRule
+    db_table = None
+
```

Comment:
```suggestion
    rule_class = GenericAdderRule
    
    __slots__ = ["db_table", "_db_connector"]    
    
    db_table : dict = None
    """your docstring here"""
    
    _db_connector : MySQLConnector
    """your docstring here"""
        
```

* attributes should be defined in class with typehint and docstring
* attributes should be defined in __slots__ for performance reasons

Reply:
I've changed it, but I could not add db_table to slots, since it had a conflict with the initialized class variable.